allensdk/brain_observatory/ecephys/ecephys_project_api/ecephys_project_lims_api.py

```diff
@@ -397,6 +497,73 @@ def get_unit_analysis_metrics(self, unit_ids=None, ecephys_session_ids=None, ses
         return response
 
 
+    def _get_template(self, name, namespace):
+        """ Identify the WellKnownFile record associated with a stimulus 
+        template and stream its data if present.
+        """
+
+        try:
+            well_known_file = build_and_execute(
+                f"""
+                select 
+                    st.well_known_file_id
+                from stimuli st
+                join stimulus_namespaces sn on sn.id = st.stimulus_namespace_id
+                where
+                    st.name = '{name}'
+                    and sn.name = '{namespace}'
+                """,
+                base=postgres_macros(),
+                engine=self.postgres_engine.select_one
+            )
+            wkf_id = well_known_file["well_known_file_id"]
+        except (KeyError, IndexError):
+            raise ValueError(f"expected exactly 1 template for {name}")
+
+        download_link = f"well_known_files/download/{wkf_id}?wkf_id={wkf_id}"
+        return self.app_engine.stream(download_link)
+
+
+    def get_natural_movie_template(self, number: int) -> Generator[bytes, None, None]:
```

Comment:
I think the return type can just be `Iterator[bytes]` or `Iterable[bytes]`. I think `Generator` should only be used when the SendType or ReturnType are not `None`.

https://docs.python.org/3.7/library/typing.html#typing.Generator
https://mypy.readthedocs.io/en/latest/kinds_of_types.html#generators